cmake/FindHIP.cmake

```diff
@@ -383,6 +408,9 @@ macro(HIP_PREPARE_TARGET_COMMANDS _target _format _generated_files _source_files
     HIP_PARSE_HIPCC_OPTIONS(HIP_HCC_FLAGS ${_hcc_options})
     HIP_PARSE_HIPCC_OPTIONS(HIP_NVCC_FLAGS ${_nvcc_options})
 
+    # Add the compile definitions
+    list(APPEND HIP_HIPCC_FLAGS "-D$<JOIN:$<TARGET_PROPERTY:${_target},COMPILE_DEFINITIONS>, -D>")
```

Comment:
Same as above. The expression assumes that ```TARGET_PROPERTY:COMPILE_DEFINITIONS``` is non-empty. It needs to be checked for non-empty, else a dangling ```-D``` would be appended to ```HIP_HIPCC_FLAGS```. See https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#output-expressions for how to do that.